value.go

```diff
@@ -53,6 +53,15 @@ func Null(iso *Isolate) *Value {
 	return iso.null
 }
 
+func NewStringFromByteArray(iso *Isolate, val []byte) (*Value, error) {
+	if iso == nil {
+		return nil, errors.New("v8go: failed to create new Value: Isolate cannot be <nil>")
+	}
+	cUint := (*C.uchar)(unsafe.Pointer(&val[0]))
+	rtnVal := C.NewValueStringFromByteArray(iso.ptr, cUint, C.int(len(val)))
+	return valueResult(nil, rtnVal)
+}
+
```

Comment:
What was your reasoning for introducing a new api instead of adding a switch case to `NewValue` that handles `[]byte` as done below for other types?

Reply:
NewValue converts Go type to V8 type. In future we might want to just implement uint8 array. (there's already a PR related to that implementation). 
But in this scenario, we just want to get a string out of byte array. so created a new api.

Reply:
As far as I understand, adding native support for typed arrays in v8go won't help here, because the JS code expects a `string` back, not a typed array. Our issue is that calling `string(bytes)` in Go doesn't create the string we need. That can also be mitigated if we do smth like:

```golang
var s string
for _, b := range bytes {
   s += string(b)
}
```
Then, `s` contains the string we want to pass back to v8.

Because this code implements a special way to handle `[]byte` when converting to an existing v8 type (string), I'd still keep it in `NewValue`.

In any cases, the library user would have to know that doing `string(bytes)` when passing data to V8 leads to erroneous results.

Reply:
@maxshirshin based on my understanding the conversion problem, doesn't originate in `string(bytes)`. It seems to be V8's `String::NewStringFromUtf8` that causes the problem: turning valid UTF-8 byte sequences into single characters rather than converting each byte to an equivalent character (as it's supposed to). In this [binary string](https://developer.mozilla.org/en-US/docs/Web/API/DOMString/Binary) case we specifically need 1 byte to result in 1 character. That's why we're using `NewFromOneByte` instead.

I'm not sure how we could implicitly distinguish between the 'binary string' and 'utf-8 string' cases in `NewValue`? As mentioned, passing a `[]byte` would preferably result in a `ArrayBuffer` or `[]Uint8` in the JS realm. If we really want to keep everything inside `NewValue`, we could make a type for it, e.g.: `type BinaryString []byte`. However, the library user would still have to know about this exception case.